flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -747,3 +747,47 @@ function retry_times() {
     echo "Command: ${command} failed ${retriesNumber} times."
     return 1
 }
+
+##################################################################################
+# Retry action until it has successfully exited and wait for condition.
+#
+# Globals:
+#   -
+# Arguments:
+#   $1 - action to run
+#   $2 - boolean function to check whether the action has been successfully done
+#   $3 - action name for logs
+#   $4 - (default: 10) max number of retries to run action until passed and check done
+#   $5 - (default: 0) backoff delay in seconds between retry attempts
+# Returns:
+#   Done or failed message
```

Comment:
Not sure if it makes sense to document the return value. Technically this function also _"returns"_ _"Retry <...> after ... seconds, ..."_ so the documentation is already inaccurate.